gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenTransformerApi.scala

```diff
@@ -63,4 +66,17 @@ abstract class GlutenTransformerApi extends ITransformerApi with Logging {
                            selectedPartitions: Array[PartitionDirectory]): Seq[InputPartition] = {
     InputPartitionsUtil.genInputPartitionSeq(relation, selectedPartitions)
   }
+
+  override def genExistsColumnProjection(selectionNode: SelectionNode,
+                                         substraitContext: SubstraitContext): ExpressionNode = {
+    // Velox will return "null" for unmatched join keys if:
+    // 1. probe-side join key is null
+    // 2. probe-side join key is not null but build-side contains null keys
```

Comment:
Is Velox's behaviour aligned with DuckDB's?

Reply:
How to tell if they are aligned?

Here's how I reproduce the failed query in velox's UT HashJoinTest.ccp, the result is empty because of the last filter.
If change the filter to `.filter("m1 is null or match is null")`, the test will pass. Is this considered as not aligned?
```
TEST_F(HashJoinTest, subquery) {
  auto l = makeRowVector(
      {"a", "b"},
      {
          makeNullableFlatVector<int32_t>({1, 1, 2, 2, 3, std::nullopt, std::nullopt, 6}),
          makeNullableFlatVector<double>({2.0, 2.0, 1.0, 1.0, 3.0, std::nullopt, 5.0, std::nullopt}),
      });

  auto r = makeRowVector(
      {"c", "d"},
      {
          makeNullableFlatVector<int32_t>({2, 2, 3, 4, std::nullopt, std::nullopt, 6}),
          makeNullableFlatVector<double>({3.0, 3.0, 2.0, 1.0, std::nullopt, 5.0, std::nullopt}),
      });

  createDuckDbTable("l", {l});
  createDuckDbTable("r", {r});

  // Plan 2 hash existence join with a filter.
  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
  auto plan = PlanBuilder(planNodeIdGenerator)
                  .values({l})
                  .hashJoin(
                      {"a"},
                      {"c"},
                      PlanBuilder(planNodeIdGenerator).values({r}).planNode(),
                      "b < d",
                      {"a", "b", "match"},
                      core::JoinType::kLeftSemiProject)
                  .project({"a", "b", "match as m1"})
                  .hashJoin(
                      {"a"},
                      {"c"},
                      PlanBuilder(planNodeIdGenerator).values({r}).planNode(),
                      "",
                      {"a", "b", "m1", "match"},
                      core::JoinType::kLeftSemiProject)
                  .filter("not m1 or not match")
                  .project({"a", "b"})
                  .planNode();

    HashJoinBuilder(*pool_, duckDbQueryRunner_, driverExecutor_.get())
      .planNode(std::move(plan))
      .referenceQuery(
          "select * from l where (not exists (select * from r where l.a = r.c and l.b < r.d)) or (not exists (select * from r where l.a = r.c))")
      .checkSpillStats(false)
      .run();
}
```

Reply:
If not, maybe we can open this issue to Velox.

Reply:
Here was a fix about `match = false` or `match = null` for another case (https://github.com/facebookincubator/velox/pull/3275). Maybe we need to check what is the expected bahaviour in Velox. If Velox behaves not as expected, this can be a issue.

Reply:
Thanks for the reference. The description "Left semi project join used to return 'match = NULL' for all rows with null join keys." in this PR also indicates that the semantics they use is different from Spark's. Also, in this case, the cardinality of join's output is correct, but the problem is how we build the filter after join.

Reply:
Does that description mean Velox's previous bahaviour was like that, but it was not correct? So they made some fixes in the mentioned PR. I'm wondering whether returning false for unmatched row is also an expected behaviour in Velox.

Reply:
Yes, I also wonder why they only fixed the special case for empty build side. I'll open an issue for discussion.